src/commcare_cloud/fab/fabfile.py

```diff
@@ -208,11 +208,29 @@ def _confirm_deploying_same_code():
     if env.deploy_metadata.current_ref_is_different_than_last:
         return
 
+    if env.code_branch == 'master':
+        branch_specific_msg = "Perhaps you meant to merge a PR or specify a --set code_branch=<branch> ?"
```

Comment:
Do you know if the tags from failed deploys get cleaned up?  There's something about `setup_release` tags and `deploy` tags, maybe the latter only happens on successful deploy?

Reply:
`deploy` only happens on successful deploys which is what last_tag looks for https://github.com/dimagi/commcare-cloud/blob/697b045adeafb192baf1ab68c6ab0dc9c9a65b50/src/commcare_cloud/fab/utils.py#L86